### PR TITLE
Add key id

### PR DESCRIPTION
**Pull request by djw8605**

### CODE REVIEW

File: src/scitokens/scitokens.py [hunk 1]
```diff
@@ -142,14 +142,15 @@ def verify(self):
         raise NotImplementedError()
 
 
-    def serialize(self, include_key=False, issuer=None, lifetime=600):
+    def serialize(self, include_key=False, issuer=None, lifetime=600, kid=None):
```
Comment (bbockelm) [inline src/scitokens/scitokens.py]: Seems awkward to have `kid` here but the key itself is specified in the constructor.

Can you move `kid` to the constructor (and spell it `key_id` instead?  We don't use abbreviations for `issuer` here...)?

File: src/scitokens/scitokens.py [hunk 2]
```diff
@@ -170,16 +171,23 @@ def serialize(self, include_key=False, issuer=None, lifetime=600):
         issue_time = int(time.time())
         exp_time = int(issue_time + lifetime)
         
-        payload = dict(self._claims)
+        # Add to validated and other claims
+        payload = dict(self._verified_claims)
+        payload.update(dict(self._claims))
```
Comment (bbockelm) [inline src/scitokens/scitokens.py]: Casting to `dict` isn't necessary, is it?

Comment (djw8605) [reply src/scitokens/scitokens.py]: The Second isn't necessary.  But the first is done so we can "copy" the _verified_claims dict and not make modifications to it.

File: src/scitokens/scitokens.py [hunk 3]
```diff
@@ -101,18 +101,20 @@ class SciToken(object):
     An object representing the contents of a SciToken.
     """
 
-    def __init__(self, key=None, parent=None, claims=None):
+    def __init__(self, key=None, key_id=None, parent=None, claims=None,  ):
```
Comment (bbockelm) [inline src/scitokens/scitokens.py]: What's with the extra `,  `?

Comment (djw8605) [reply src/scitokens/scitokens.py]: Good catch

**Pull request merged**